ts/features/payments/checkout/saga/networking/handleWalletPaymentAuthorization.ts

```diff
@@ -67,26 +66,30 @@ export function* handleWalletPaymentAuthorization(
       action
     )) as SagaCallReturnType<typeof requestTransactionAuthorization>;
 
-    yield* put(
-      pipe(
-        requestTransactionAuthorizationResult,
-        E.fold(
-          error =>
-            paymentsStartPaymentAuthorizationAction.failure({
-              ...getGenericError(new Error(readablePrivacyReport(error)))
-            }),
+    if (E.isLeft(requestTransactionAuthorizationResult)) {
+      yield* put(
+        paymentsStartPaymentAuthorizationAction.failure({
+          ...getGenericError(
+            new Error(readablePrivacyReport(requestTransactionAuthorizationResult.left))
+          )
+        })
+      );
+      return;
+    }
 
-          res => {
-            if (res.status === 200) {
-              return paymentsStartPaymentAuthorizationAction.success(res.value);
-            }
-            return paymentsStartPaymentAuthorizationAction.failure({
-              ...getGenericError(new Error(`Error: ${res.status}`))
-            });
-          }
+    if (requestTransactionAuthorizationResult.right.status === 200) {
+      yield* put(
+        paymentsStartPaymentAuthorizationAction.success(
+          requestTransactionAuthorizationResult.right.value
         )
-      )
-    );
+      );
+    } else if (requestTransactionAuthorizationResult.right.status !== 401) {
+      yield* put(
+        paymentsStartPaymentAuthorizationAction.failure({
+          ...getGenericError(new Error(`Error: ${requestTransactionAuthorizationResult.right.status}`))
+        })
+      );
+    }
```

Comment:
Why did you remove the `pipe` here? We could achieve the same result still using a functional approach:

```typescript
 yield pipe(
      requestTransactionAuthorizationResult,
      E.fold(
        function* (error) {
          yield* put(
            paymentsStartPaymentAuthorizationAction.failure(
              getGenericError(new Error(readablePrivacyReport(error)))
            )
          );
        },
        function* ({ status, value }) {
          switch (status) {
            case 200:
              yield* put(
                paymentsStartPaymentAuthorizationAction.success(value)
              );
              break;
            case 401:
              // 401 status code will be handled by LV authentication flow
              break;
            default:
              yield* put(
                paymentsStartPaymentAuthorizationAction.failure(
                  getGenericError(new Error(`Error: ${status}`))
                )
              );
              break;
          }
        }
      )
    );
  ```

Reply:
To be honest, I don't like the way to manage these statuses through the use of `pipe` and fp-ts as you know I'm not a huge fan of it, I think that sometimes can be useful but in this scenario, it adds a lot of boilerplate and not-so-understandable code. I prefer to handle them as well as all the other domains.

Regarding the explanation of why the 401 is not handled I agree with you and I'm going to add some comments

Reply:
Comments added into commit 121028e0659cfe533ab79de56a880deb34b3013b 

@mastro993

Reply:
Comments added into commit 121028e0659cfe533ab79de56a880deb34b3013b 

@mastro993